apps/nextjs/src/components/AppComponents/Chat/chat-lessonPlanDisplay.tsx

```diff
@@ -41,7 +48,10 @@ const displayStyles = cva(
   "relative flex flex-col space-y-10 px-14 pb-28 opacity-100 sm:px-24 ",
 );
 
-const organiseSections = [
+const organiseSections: {
```

Comment:
I'm still very unconvinced by this method. It assumes that the LLM always responds in chunks that match what it is expecting. If it sends back three sections when you're expecting four, you end up with a broken section that never loads and you can't interact with. I would vote for removing this and just using the sections that are present in the lesson plan.

Reply:
It does seem slightly brittle to me. It's almost like you'd want the LLM to return with the names of the sections it's about to stream upfront in order for it to work nicely